pkg/goDB/DBWorkManager.go

```diff
@@ -318,20 +350,22 @@ func (w *DBWorkManager) readBlocksAndEvaluate(workload DBWorkload, resultMap has
 
 		key, comparisonValue := v4Key, v4ComparisonValue
 		startEntry, isIPv4 := 0, true // TODO: Support traversal of IPv4 / IPv6 only if there's a matching condition
+		finalMap := resultMap.V4Map
 		for i := startEntry; i < numEntries; i++ {
 
-			// When reaching the v4/v6 mark, we switch to the IPv6 key
+			// When reaching the v4/v6 mark, we switch to the IPv6 key / submap
 			if i == int(numV4Entries) {
 				key, comparisonValue = v6Key, v6ComparisonValue
+				finalMap = resultMap.V6Map
```

Comment:
Then where do the `V4` entries go to? Are they still part of `finalMap`?

Reply:
Jup, that's just flipping the pointer from v4 to v6 (see line above: `finalMap := resultMap.V4Map`) without having to check on each entry (because it's sorted / separated anyway).
However, with the extension of the `AggFlowMap` I did further up I can of course now get rid of that und just use the boolean flag, directly inserting into the wrapped map instead of manually handling IPv4 / IPv6 here. Will adapt accordingly.